src/api/supabase/queries/user_queries.ts

```diff
@@ -16,87 +16,166 @@ const supabaseApiKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;
 // Initialize the Supabase client
 const supabase = createClient(supabaseUrl ?? '', supabaseApiKey ?? '');
 
-export async function fetchUserData(): Promise<
-  PostgrestSingleResponse<User[]> | { data: never[]; error: PostgrestError }
-> {
-  try {
-    const { data: users, error } = await supabase
-      .from('profiles')
-      .select('*')
-      .single();
-
-    if (error) {
-      console.error('Error fetching data:', error);
-      return { data: [], error };
-    }
+export async function fetchUserData() {
+  const { data, error }: { data: User[] | null; error: PostgrestError | null } =
+    await supabase.from('profiles').select('*');
 
-    return { data: users } as PostgrestSingleResponse<User[]>;
-  } catch (error) {
-    console.error('Error:', error);
-    throw error;
+  if (error) {
+    throw new Error(`An error occured when trying to read profiles: ${error}`);
+  } else {
+    return data;
   }
 }
 
 export async function fetchUserByUUID(uuid: string) {
-  try {
-    const { data: user, error } = await supabase
-      .from('profiles')
-      .select('*')
-      .eq('user_id', uuid)
-      .single();
-
-    if (error) {
-      console.error('Error fetching user data:', error);
-    }
+  const { data, error } = await supabase
+    .from('profiles')
+    .select('*')
+    .eq('user_id', uuid)
+    .single();
+  if (error) {
+    throw new Error(`An error occured when trying to read profiles: ${error}`);
+  } else {
+    return data;
+  }
+}
+
+export async function fetchUserCart(
```

Comment:
i'd say for all queries that are rn taking the `userId` as a parameter, you can remove it. i don't think yall are using an auth/profile context, so i'd say when you're calling these queries from your components, it's probably easier to just omit getting the `userId` there and passing it to the queries. rather i'd say it'd be easier to just use the `supabase.auth.getUser()` function (which returns current logged in user which makes sense b/c all of these queries should only be available to person who's logged in at the moment) inside these queries and you can get the `userId` off of the object returned here.